src/views/Products.jsx

```diff
@@ -1,7 +1,104 @@
-import React from 'react'
+/* eslint-disable no-unused-vars */
+import React, { useState, useEffect } from 'react'
+import PropTypes from 'prop-types'
+import ProductsSidebar from '../components/ProductsSidebar'
+import ProductCategories from '../mocks/en-us/product-categories.json'
+import ProductsJson from '../mocks/en-us/products.json'
+import '../styles/Products.scss'
+import ProductItem from '../components/FeaturedProducts/ProductItem'
+import ProductsPagination from '../components/ProductsPagination'
+import Spinner from '../components/Spinner'
+
+export default function Products ({ onHandleLinkPage }) {
+  const [isLoading, setIsLoading] = useState(true)
+  const [filters, setFilters] = useState([])
+  const [filteredProducts, setFilteredProducts] = useState(ProductsJson.results)
+
+  const isInFilters = (value) => {
+    return filters.indexOf(value) !== -1
+  }
+
+  const filterProducts = () => {
+    const products = ProductsJson.results.filter((fProduct) => {
+      return filters.includes(fProduct.data.category.id)
+    })
+    setFilteredProducts(products)
+  }
+
+  const resetLoading = () => {
+    setTimeout(() => {
+      setIsLoading(false)
+    }, 2000)
+  }
+
+  const handleFilters = (e) => {
+    const { checked, value } = e.target
+    setIsLoading(true)
+
+    if (checked) {
+      filters.push(value)
+      setFilters(filters)
+    } else if (isInFilters(value)) {
+      filters.splice(filters.indexOf(value), 1)
+      setFilters(filters)
+    }
+
+    resetLoading()
+
+    if (filters.length === 0) {
+      setFilteredProducts(ProductsJson.results)
+      return null
+    }
+    filterProducts()
```

Comment:
this could also be refactored because the function `filterProducts` is only being used here, having a piece of the code in one part and the rest in another makes it hard to reason about

The part of the code that updates the filtered products should be in a separate `useEffect` that fires when the filters change, it could be something like:

```jsx
useEffect(() => {
  const newProducts = filters.length === 0
    ? ProductsJson.results
    : ProductsJson.results.filter(
      p => filters.includes(p.data.category.id)
    )
  setFilteredProducts(newProducts)
}, [filters, ProductsJson.results])
```